analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -246,28 +260,31 @@ class Pub(
 
             val parsePackagesResult = parseInstalledPackages(lockFile, labels, workingDir)
 
-            val gradleDefinitionFiles = gradleDefinitionFilesForPubDefinitionFiles.getValue(definitionFile).toList()
-
-            if (gradleFactory != null) {
-                if (gradleDefinitionFiles.isNotEmpty()) {
-                    val gradleDependencies = gradleDefinitionFiles.map {
-                        PackageManagerDependencyHandler.createPackageManagerDependency(
-                            packageManager = gradleFactory.type,
-                            definitionFile = VersionControlSystem.getPathInfo(it).path,
-                            scope = "releaseCompileClasspath",
-                            linkage = PackageLinkage.PROJECT_STATIC
-                        )
-                    }.toSortedSet()
+            if (!pubDependenciesOnly) {
+                if (gradleFactory != null) {
+                    val gradleDefinitionFiles = gradleDefinitionFilesForPubDefinitionFiles
+                        .getValue(definitionFile)
+                        .toList()
+                    if (gradleDefinitionFiles.isNotEmpty()) {
```

Comment:
Nit: There should now be an empty line before this one to separate from the wrapped block above.